<a href="https://colab.research.google.com/github/lkivumbi/Project-2-Part-time-DataScience-Student-Refactory-Uganda/blob/main/Project2_part1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Assignment: Choose and describe 2 potential datasets for your second project.
You will have a lot more freedom in your second project than in your first project. This is because we want you to have a project in your portfolio that interests you or relates to the industry you would like to work in.

Your task for this week is to propose two possible datasets you would like to work with for Project 2 and indicate which is your first and which is your second choice.

You can choose datasets from either:

1. This list of pre-approved datasets and complete the tasks listed below for both of them
**Pre-approved Datasets:**
- Adult income dataset
- Car Insurance Data
- Metabolic Syndrome Prediction - dataset by informatics-edu
- Pump It Up Challenge: Driven Data
- Cirrhosis Prediction Dataset
- Spanish Wine Quality Dataset


2. **Other dataset(s) that align with an interest or field of expertise you already have.**

If you choose to source your own data, please be sure it meets the below requirements:
- has a clear target or labels to predict has potential for interesting analysis
- the data must be available for use (it is your responsibility to ensure that - the license states that you are able to use it.)
- the data must be appropriate for a professional environment
- the data must not contain personal information
- the data must not be a dataset used for any assignment, lecture, or task from the course
- the data must not be a time series dataset. You will be able to identify these because each row will represent a moment or interval of time. These kinds of datasets follow special rules and are not appropriate for the kind of machine learning you have learned in this stack.
- is reasonable to work with during the time you have

**A great dataset would also have:**

- 8-25 columns (features)
- 1000 - 100,000 rows (instances)
- both numeric and categorical features
- At least some missing features in both categorical and numeric columns, so you can show off your data-cleaning skills!

Here are some resources to help you find great datasets:

- The 50 Best Free Datasets for Machine Learning
- 21 Places to Find Free Datasets for Data Science Projects
- Best Public Datasets for Machine Learning and Data Science
- Top 20+ Datasets for Machine Learning and Statistics Projects
- 70+ Machine Learning Datasets & Project Ideas
- DataRobot Pathfinder

You may choose a regression or classification problem for each proposed data set, but you might consider a classification project since you already have a regression project in your portfolio. This way you can show your skills with both!

If you choose 2 datasets from outside of the pre-approved list, at least one of the datasets you choose must be approved by the instructor before you can use it for your project.

##For this Task:
Whether you chose 2 pre-approved datasets or 2 datasets from another source or one of each:

Create a notebook where you have uploaded and shown the .head() of each of your data sets. For each of the proposed datasets, include the following information in text cells:

First choice: dataset 1
1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person? A business? An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

Second choice: dataset 2
1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person? A business? An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

Create a NEW GitHub repository for this project, with a readme, and add your notebook to that new repository. Submit the link below.


In [1]:
#importing libraries
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd

In [3]:
#loading the data
fpath = '/content/drive/MyDrive/StudentFolders16wk24wk/CodingDojo/02-IntroML/Week07/Data/wines_SPA.csv'
fname = '/content/drive/MyDrive/StudentFolders16wk24wk/CodingDojo/02-IntroML/Week07/Data/cirrhosis.csv'
df1 = pd.read_csv(fpath) #where this dataframe will represent my first choice dataset
df2 = pd.read_csv(fname) #where this dataframe will be my second choice dataframe
display(df1.head())
df2.head()

winery           wine  year  rating  num_reviews country  \
0  Teso La Monja          Tinto  2013     4.9           58  Espana   
1         Artadi  Vina El Pison  2018     4.9           31  Espana   
2   Vega Sicilia          Unico  2009     4.8         1793  Espana   
3   Vega Sicilia          Unico  1999     4.8         1705  Espana   
4   Vega Sicilia          Unico  1996     4.8         1309  Espana   

             region   price                  type  body  acidity  
0              Toro  995.00              Toro Red   5.0      3.0  
1    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0

ID  N_Days Status             Drug    Age Sex Ascites Hepatomegaly Spiders  \
0   1     400      D  D-penicillamine  21464   F       Y            Y       Y   
1   2    4500      C  D-penicillamine  20617   F       N            Y       Y   
2   3    1012      D  D-penicillamine  25594   M       N            N       N   
3   4    1925      D  D-penicillamine  19994   F       N            Y       Y   
4   5    1504     CL          Placebo  13918   F       N            Y       Y   

  Edema  Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos    SGOT  \
0     Y       14.5        261.0     2.60   156.0    1718.0  137.95   
1     N        1.1        302.0     4.14    54.0    7394.8  113.52   
2     S        1.4        176.0     3.48   210.0     516.0   96.10   
3     S        1.8        244.0     2.54    64.0    6121.8   60.63   
4     N        3.4        279.0     3.53   143.0     671.0  113.15   

   Tryglicerides  Platelets  Prothrombin  Stage  
0          172.0      190.0         12.2    4.0  
1           88.0      221.0         10.6    3.0  
2           55.0      151.0         12.0    4.0  
3           92.0      183.0         10.3    4.0  
4           72.0      136.0         10.9    3.0

In [5]:
#understanding dataset 1
df1.shape

(7500, 11)

In [7]:
#information of dataset 1
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7500 entries, 0 to 7499
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   winery       7500 non-null   object 
 1   wine         7500 non-null   object 
 2   year         7498 non-null   object 
 3   rating       7500 non-null   float64
 4   num_reviews  7500 non-null   int64  
 5   country      7500 non-null   object 
 6   region       7500 non-null   object 
 7   price        7500 non-null   float64
 8   type         6955 non-null   object 
 9   body         6331 non-null   float64
 10  acidity      6331 non-null   float64
dtypes: float64(4), int64(1), object(6)
memory usage: 644.7+ KB


In [9]:
#understanding dataset2
df2.shape

(418, 20)

In [10]:
#information on dataset 2
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 20 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   ID             418 non-null    int64  
 1   N_Days         418 non-null    int64  
 2   Status         418 non-null    object 
 3   Drug           312 non-null    object 
 4   Age            418 non-null    int64  
 5   Sex            418 non-null    object 
 6   Ascites        312 non-null    object 
 7   Hepatomegaly   312 non-null    object 
 8   Spiders        312 non-null    object 
 9   Edema          418 non-null    object 
 10  Bilirubin      418 non-null    float64
 11  Cholesterol    284 non-null    float64
 12  Albumin        418 non-null    float64
 13  Copper         310 non-null    float64
 14  Alk_Phos       312 non-null    float64
 15  SGOT           312 non-null    float64
 16  Tryglicerides  282 non-null    float64
 17  Platelets      407 non-null    float64
 18  Prothrombi

#First choice Dataset: Spanish Wine Dataset

##Source
* This dataset has been sourced from kaggle with the link provided here https://www.kaggle.com/datasets/fedesoriano/spanish-wine-quality-dataset.

##Brief description:
This dataset is related to red variants of spanish wines. The dataset describes several popularity and description metrics their effect on it's quality. The dataset has 11 features and 7500 rows or entries.

##what is the target?
According to the source, this dataset can be used to determine either the price, or quality of the wine. for this project we shall use price as the target.

##what does one row represent(A person, A business, an event, aproduct)
One row here represents a type of wine and its different features like year of manufacture, winery it was made at among others

##is this a classification or regression problem?
For this project this will be a regression problem

##How many features does the data have
this dataset has 11features.

##how many rows are in the dataset
this dataset has 7500 rows

##What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
I foresee challenges of in imputing missing values, dealing with inconsistent values and wrong datatype features.

#Second choice: Cirrhosis Prediction Dataset

##Source
* This dataset has been sourced from kaggle with the link provided here https://www.kaggle.com/fedesoriano/cirrhosis-prediction-dataset.

##Brief description:
Cirrhosis is a late stage of scarring (fibrosis) of the liver caused by many forms of liver diseases and conditions, such as hepatitis and chronic alcoholism. The following data contains the information collected from the Mayo Clinic trial in primary biliary cirrhosis (PBC) of the liver conducted between 1974 and 1984.

##what is the target?
the target for this dataset would be the status

##what does one row represent(A person, A business, an event, aproduct)
One row here would represent a patient a person and there different describing features.

##is this a classification or regression problem?
This is a classification problem

##How many features does the data have
this dataset has 20 features

##how many rows are in the dataset
this dataset has 418 rows

##What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
I foresee challenges of imputing missing values, dealing with inconsistent values and wrong datatypes.